{{cookiecutter.project_slug}}/.gitlab-ci.yml

```diff
@@ -0,0 +1,131 @@
+# templates
+# ------------
+
+.deploy-template:
+  image: alpine:latest
+  script:
+    - echo "Deploying to ${CI_ENVIRONMENT_NAME}"
+  variables:
+    GIT_STRATEGY: none  # prevents fetching repo (we only need the artifact here)
+
+# -------------
+# Pipeline Definition
+
+variables:
+  PIP_CACHE_DIR: $CI_PROJECT_DIR/.cache/pip
+  {%- if cookiecutter.package_manager == 'conda' %}
+  CONDA_PKGS_DIRS: $CI_PROJECT_DIR/.cache/conda
+  {%- elif cookiecutter.package_manager == 'poetry' %}
+  POETRY_VERSION: 1.0.5
+  POETRY_CACHE_DIR: $CI_PROJECT_DIR/.cache/poetry
+  {%- endif %}
+
+cache: &python_cache
+  paths:
+    - $PIP_CACHE_DIR
+    {%- if cookiecutter.package_manager == 'conda' %}
+    - $CONDA_PKGS_DIRS/*.tar.bz2
+    - $CONDA_PKGS_DIRS/urls.txt
+    {%- elif cookiecutter.package_manager == 'poetry' %}
+    - $POETRY_CACHE_DIR
+    {%- endif %}
+
+stages:
+  - build-wheel
+  - test-unit
+  - deploy-nonprod
+  - deploy-prod
+
+# Build and Test
+{% if cookiecutter.package_manager == 'poetry' -%}
+build-wheel:
+  image: python:3.7-stretch
+  stage: build-wheel
+  artifacts:
+    name: 'app-wheel'
+    paths:
+      - dist/{{ cookiecutter.module_name }}-*.whl
+    expire_in: 1 week
+  before_script:
+    - pip install poetry==$POETRY_VERSION
+  script:
+    - poetry build -f wheel
+{% else -%}
+build-wheel:
+  image: python:3.7-alpine
+  stage: build-wheel
+  artifacts:
+    name: 'app-wheel'
+    paths:
+      - dist/{{ cookiecutter.module_name }}-*.whl
+    expire_in: 1 week
```

Comment:
just one week? I think we can afford the storage space and wheels are usually less than 1mb, something like 3-6 months would be fine for me

Reply:
Actually I just put in any number here just to show the expiration capability of gitlab ci. 
How long you store it should be dependend on the use-case and where gitlab is hosted. In general I agree though that artifacts should probably be stored longer than 1 week so to provide a best practice I will increase the time :)